src/visit_vtk/full/vtkLinesFromOriginalCells.C

```diff
@@ -255,6 +258,22 @@ int vtkLinesFromOriginalCells::RequestData(
               } // if unique edge
           } // for all edge points
       }//for all edges of cell
+
```

Comment:
Good point. I will double-check whether VTK_LINE primitives are still part of the output in this case.

Reply:
vtkExtractEdges filter (used in the case where no avtOriginalCellNumbers array is present) does NOT preserve VTK_LINE primitives.  I'm not sure whether its better to sublcass vtkExtractEdges to preserve the lines, or simply resurrect them in our filter after vtkExtractEdges is done.

Reply:
Kinda depends I think on what vtkExtractEdges is intended to do.

From VTK documentation...
>extract cell edges from any type of data
>vtkExtractEdges is a filter to extract edges from a dataset. Edges are extracted as lines or polylines.

It would seem that an edge primitive, `VTK_LINE`, is itself an edge and so *extracting* it from itself is kinda sorta the identity. In other words, I think that is a bug in vtkExtractEdges and that it should return any `VTK_LINE` primitives that it was given.

Reply:
Found this email thread:
https://public.kitware.com/pipermail/vtk-developers/2008-May/020685.html
So, VTK_LINE cells do not have edges, so the behavior of vtkExtractEdges is correct for its purposes.

Reply:
I disagree with W's remarks in that email. It may be a correct statement about what he expects the class `vtkExtractEdges` to do. Thats fine, it is probably better named `vtkExtractEdgesFrom2or3DCells` but it I find it awkward to try to support that behavior by backing it up with the mathemtical/topological statement that an edge has no edges or that a face has no faces. If I give you a poly-line (by defn a 1D cell) or a triangle-strip (by defn a 2D cell) and ask you to give me back the *edges* or the *faces*, are you going to return the empty set? I don't think so. Now, make each of those *cells* an entity of size 1 (e.g. a polyline of a single line or a traingle-strip of a single triangle). Do you return empty set then? I don't think so. An edge has *edges* because it *is* an edge. A face has *faces* because it *is* a face. Now, that may not be the desired behavior of `vtkExtractEdges` but to try to justify that behavior with awkard topological statements as opposed to *that is just the way this class is intended to behave* is hard to swallow.

Reply:
So, whether or not we agree with it, the intentional behavior in VTK appears to be that given a cell that is comprised of line primitive(s), querying 'number of edges' for that cell will always return 0, and any code that loops on the number of edges of a cell will fail to process lines.  We can implement our own edge extractor that will processes lines as edges.